source/tracing.mimic.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) bwplotka/mimic Authors
+// Licensed under the Apache License 2.0.
+
+package main
+
+import (
+	"time"
+
+	"github.com/bwplotka/mimic/lib/abstr/kubernetes/volumes"
+	"github.com/go-openapi/swag"
+	"github.com/grafana/tempo/cmd/tempo/app"
+	"github.com/grafana/tempo/modules/compactor"
+	"github.com/grafana/tempo/modules/distributor"
+	"github.com/grafana/tempo/modules/ingester"
+	"github.com/grafana/tempo/modules/storage"
+	"github.com/grafana/tempo/tempodb"
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/backend/local"
+	"github.com/grafana/tempo/tempodb/encoding"
+	"github.com/grafana/tempo/tempodb/pool"
+	"github.com/grafana/tempo/tempodb/wal"
+	"github.com/weaveworks/common/server"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func getTempo(name string) (appsv1.StatefulSet, corev1.Service, corev1.ConfigMap) {
+	const (
+		configVolumeName  = "tempo-config"
+		configVolumeMount = "/etc/tempo"
+		dataPath          = "/data"
+		httpPort          = 9090
+	)
+	configAndMount := volumes.ConfigAndMount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   configVolumeName,
+			Labels: map[string]string{selectorName: name},
+		},
+		VolumeMount: corev1.VolumeMount{Name: configVolumeName, MountPath: configVolumeMount},
+		Data: map[string]string{
+			"tempo.yaml": EncodeYAML(app.Config{
+				AuthEnabled: false,
+				Server: server.Config{
+					HTTPListenPort: httpPort,
+				},
+				Ingester: ingester.Config{
+					MaxTraceIdle:     10 * time.Second,
+					MaxBlockBytes:    1e6,
+					MaxBlockDuration: 5 * time.Minute,
+				},
+				Distributor: distributor.Config{
+					Receivers: map[string]interface{}{
+						"jaeger": map[string]interface{}{
+							"protocols": map[string]interface{}{
+								"grpc":        nil,
+								"thrift_http": nil,
+							},
+						},
+						"otlp": map[string]interface{}{
+							"protocols": map[string]interface{}{
+								"grpc": nil,
+							},
+						},
+					},
+				},
+				Compactor: compactor.Config{
+					Compactor: tempodb.CompactorConfig{
+						MaxCompactionRange:      1 * time.Hour,
+						MaxBlockBytes:           1e6,
+						BlockRetention:          1 * time.Hour,
+						CompactedBlockRetention: 10 * time.Minute,
+					},
+				},
+				StorageConfig: storage.Config{
+					Trace: tempodb.Config{
+						Backend: "local",
+						Block: &encoding.BlockConfig{
+							BloomFP:              .05,
+							IndexDownsampleBytes: 1000,
```

Comment:
This is rather low. We use 1MB with zstd encoding. It controls the total bytes that a single index record references in the backend.